exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/DatabaseConfig.kt

```diff
@@ -115,20 +136,52 @@ class DatabaseConfig private constructor(
          */
         @ExperimentalKeywordApi
         var preserveKeywordCasing: Boolean = true,
-    )
+    ) {
+        @Deprecated(
+            message = "This property will be removed in future releases",
+            replaceWith = ReplaceWith("defaultMaxAttempts"),
+            level = DeprecationLevel.WARNING
+        )
+        var defaultRepetitionAttempts: Int
+            get() = defaultMaxAttempts
+            set(value) { defaultMaxAttempts = value }
+
+        @Deprecated(
+            message = "This property will be removed in future releases",
+            replaceWith = ReplaceWith("defaultMinRetryDelay"),
+            level = DeprecationLevel.WARNING
+        )
+        var defaultMinRepetitionDelay: Long
+            get() = defaultMinRetryDelay
+            set(value) { defaultMinRetryDelay = value }
+
+        @Deprecated(
+            message = "This property will be removed in future releases",
+            replaceWith = ReplaceWith("defaultMaxRetryDelay"),
+            level = DeprecationLevel.WARNING
+        )
+        var defaultMaxRepetitionDelay: Long
+            get() = defaultMaxRetryDelay
+            set(value) { defaultMaxRetryDelay = value }
+    }
 
     companion object {
         operator fun invoke(body: Builder.() -> Unit = {}): DatabaseConfig {
             val builder = Builder().apply(body)
+            require(builder.defaultMaxAttempts > 0) { "defaultMaxAttempts must be set to perform at least 1 attempt." }
+
```

Comment:
This check could also be put in 2 other locations:
1. `DatabaseConfig.init()`
2. Directly in a setter for `Builder.defaultMaxAttempts`. This would require the property being pulled out of the class constructor and into the class body.

This is related to properties set in the following way:
```kt
Database.connect(
    "jdbc:h2:mem:db1;DB_CLOSE_DELAY=-1;", "org.h2.Driver",
    databaseConfig = DatabaseConfig {
        defaultMaxAttempts = 1
    }
)
```

If either option is preferable, please let me know.